### PR TITLE
[train] fix the example

### PR DESCRIPTION
**Pull request by JiahaoYao**
```python
train_dataset = ray.data.from_items(
    [{"x": i, "y": i} for i in range(3)])
```

gives the error of 

```
  File "/Users/jimmy/opt/anaconda3/envs/dl/lib/python3.7/site-packages/ray/ml/trainer.py", line 381, in _trainable_func
    super()._trainable_func(self._merged_config, reporter, checkpoint_dir)
  File "/Users/jimmy/opt/anaconda3/envs/dl/lib/python3.7/site-packages/ray/tune/function_runner.py", line 639, in _trainable_func
    output = fn()
  File "/Users/jimmy/opt/anaconda3/envs/dl/lib/python3.7/site-packages/ray/ml/trainer.py", line 356, in train_func
    trainer.training_loop()
  File "custom_trainer.py", line 22, in training_loop
    pred = self.model(X)
  File "/Users/jimmy/opt/anaconda3/envs/dl/lib/python3.7/site-packages/torch/nn/modules/module.py", line 1110, in _call_impl
    return forward_call(*input, **kwargs)
  File "/Users/jimmy/opt/anaconda3/envs/dl/lib/python3.7/site-packages/torch/nn/modules/linear.py", line 103, in forward
    return F.linear(input, self.weight, self.bias)
RuntimeError: expected scalar type Float but found Long
```
and 

```python
train_dataset = ray.data.from_items(
    [{"x": float(i), "y": float(i)} for i in range(3)])
```

gives the error of 

```python
(TrainTrainable pid=49966) 2022-06-10 09:48:06,523      ERROR function_runner.py:286 -- Runner Thread raised error.
(TrainTrainable pid=49966) Traceback (most recent call last):
(TrainTrainable pid=49966)   File "/Users/jimmy/opt/anaconda3/envs/dl/lib/python3.7/site-packages/ray/tune/function_runner.py", line 277, in run
(TrainTrainable pid=49966)     self._entrypoint()
(TrainTrainable pid=49966)   File "/Users/jimmy/opt/anaconda3/envs/dl/lib/python3.7/site-packages/ray/tune/function_runner.py", line 352, in entrypoint
(TrainTrainable pid=49966)     self._status_reporter.get_checkpoint(),
(TrainTrainable pid=49966)   File "/Users/jimmy/opt/anaconda3/envs/dl/lib/python3.7/site-packages/ray/util/tracing/tracing_helper.py", line 462, in _resume_span
(TrainTrainable pid=49966)     return method(self, *_args, **_kwargs)
(TrainTrainable pid=49966)   File "/Users/jimmy/opt/anaconda3/envs/dl/lib/python3.7/site-packages/ray/ml/trainer.py", line 381, in _trainable_func
(TrainTrainable pid=49966)     super()._trainable_func(self._merged_config, reporter, checkpoint_dir)
(TrainTrainable pid=49966)   File "/Users/jimmy/opt/anaconda3/envs/dl/lib/python3.7/site-packages/ray/tune/function_runner.py", line 639, in _trainable_func
(TrainTrainable pid=49966)     output = fn()
(TrainTrainable pid=49966)   File "/Users/jimmy/opt/anaconda3/envs/dl/lib/python3.7/site-packages/ray/ml/trainer.py", line 356, in train_func
(TrainTrainable pid=49966)     trainer.training_loop()
(TrainTrainable pid=49966)   File "custom_trainer.py", line 22, in training_loop
(TrainTrainable pid=49966)     pred = self.model(X)
(TrainTrainable pid=49966)   File "/Users/jimmy/opt/anaconda3/envs/dl/lib/python3.7/site-packages/torch/nn/modules/module.py", line 1110, in _call_impl
(TrainTrainable pid=49966)     return forward_call(*input, **kwargs)
(TrainTrainable pid=49966)   File "/Users/jimmy/opt/anaconda3/envs/dl/lib/python3.7/site-packages/torch/nn/modules/linear.py", line 103, in forward
(TrainTrainable pid=49966)     return F.linear(input, self.weight, self.bias)
(TrainTrainable pid=49966) RuntimeError: expected scalar type Float but found Double
```

use `float32` fixes the issue

### CODE REVIEW

**Comment by matthewdeng** [comment]
Can you close this PR and open a new one with this just this change?

**Comment by matthewdeng** [comment]
@clarkzinzow  FYI

File: python/ray/train/base_trainer.py [hunk 1]
```diff
@@ -122,8 +122,9 @@ def training_loop(self):
     .. code-block:: python
 
         import ray
+        import numpy as np
         train_dataset = ray.data.from_items(
-            [{"x": i, "y": i} for i in range(3)])
+            [{"x": np.float32(i), "y": np.float32(i)} for i in range(3)])
```
Comment (clarkzinzow) [inline python/ray/train/base_trainer.py:127]: Could you instead move this cast to the `to_torch()` call, using `label_column_dtype` and `feature_column_dtypes`? https://docs.ray.io/en/master/data/package-ref.html#ray.data.Dataset.to_torch

Comment (JiahaoYao) [reply python/ray/train/base_trainer.py:127]: this actually sounds good to me!

Comment (JiahaoYao) [reply python/ray/train/base_trainer.py:127]: this example also shows up here: https://github.com/ray-project/ray/blob/master/python/ray/air/examples/custom_trainer.py#L56

**Comment by JiahaoYao** [comment]
Question: shall we merge this or close this ticket? i am okay with either.

**Pull request closed without merging**